### DISCUSSION TITLE
Welcome to daftlistings Discussions!

### DISCUSSION THREAD

**Discussion by AnthonyBloomer** [AnthonyBloomer/daftlistings#85] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sdfordham** [comment]
Can we get a dev branch to start the repairs etc.
I did try.. incidentally it's better I don't have push privs.

**Reply by AnthonyBloomer** [reply]
Good idea. I created a dev branch there.

**Comment by EricNeville** [comment]
Hi, I've been trying to get this working as a user and it seems that the _ranges and _section parts are not doing anything in the search. The result is that I can filter by property type, but not by number of beds or price so I have just been doing this manually on the search results instead. Is this a known issue? I didn't find it mentioned anywhere else in the repo.

**Reply by sdfordham** [reply]
@EricNeville Can you create a new issue with a code snippet indicating the problem, thanks.

**Comment by EricNeville** [comment]
Not sure if this is mentioned anywhere else, I didn't see it in dependencies so apologies if this is already acknowledged somewhere. I think that the format of the function annotations used in the daft file are incompatible with python versions before 3.9. In particular the line:

` def search(self, max_pages: Optional[int] = None) -> list[Listing]:`

causes a problem as the type list can't be indexed. It's an easy fix to just remove it or I think the recommended way to use this annotation is to use List from the typing module:

```
from typing import List
def search(self, max_pages: Optional[int] = None) -> List[Listing]:
```

Maybe this is worth acknowledging somewhere if it is not already.

**Reply by AnthonyBloomer** [reply]
Thanks, I noticed this earlier. I'm using Python 3.9 and haven't done too much testing with prior versions of Python. Would you like to add a PR with the fix for this?

**Reply by EricNeville** [reply]
Will do

**Comment by AnnaOS-82** [comment]
Hi,
I was wondering if there is any way to also get the description of listing. The reason I ask is that I would like to filter only listings with specific keyword.

Thanks in advance
Anna

**Reply by sdfordham** [reply]
The description seems to not always be in the same place. Something like this works most of the time:

````python
daft = Daft()
daft.set_location(Location.DUBLIN)
daft.set_search_type(SearchType.RESIDENTIAL_RENT)
daft.set_property_type(PropertyType.APARTMENT)

listings = daft.search(max_pages=1)

for i, listing in enumerate(listings):
    if "location" in listing._result["prs"].keys():
        print(f"Listing #{i}:\n", listing._result["prs"]["location"])
````

Someone should have a look to try and figure out all the different location in the `dict` that it is found. Can you open an issue about this
